integration-tests/contract-permits.spec.ts

```diff
@@ -0,0 +1,242 @@
+      .catch(Error) ;
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      done();
+    });
+  });
+
+  test('Permit can be submitted and used', async (done) => {
+
+      const op = await Tezos.contract.originate({
+        code: permit_admin_42,
+        storage:
+        {
+        0: new MichelsonMap(),
+        1: 0,
+        2: 'tz1bDCu64RmcpWahdn9bWrDMi6cu7mXZynHm'
+        },
+        });
+        await op.confirmation();
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        const contract = await op.contract();
+        expect(op.status).toEqual('applied')
+
+        const permit_parameter = 'Pair "edpkuPTVBFtbYd6gZWryXypSYYq6g7FvyucwphoU78T1vmGkbhj6qb" (Pair "edsigtfkWys7vyeQy1PnHcBuac1dgj2aJ8Jv3fvoDE5XRtxTMRgJBwVgMTzvhAzBQyjH48ux9KE8jRZBSk4Rv2bfphsfpKP3ggM" 0x0f0db0ce6f057a8835adb6a2c617fd8a136b8028fac90aab7b4766def688ea0c)'
+
+        //submit the permit_parameter to the contract
+        Tezos.contract
+        .at(contract.address)
+        .then((c) => {
+          return c.methods.permit(permit_parameter).send();
+        })
+        .then(async (op) => {
+            await op.confirmation(3);
+            return op.hash;
+          })
+          .catch(Error) ;
+          expect(op.hash).toBeDefined();
+          expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+
+        //submit 42 to the wrapped entrypoint
+        Tezos.contract
+        .at(contract.address)
+        .then((c) => {
+          return c.methods.wrapped("42").send();
+        })
+        .then(async (op) => {
+            await op.confirmation(3);
+            return op.hash;
+          })
+        .catch(Error) ;
+        expect(op.hash).toBeDefined();
+        expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+        done();
+      });
+
+  test('Deploy a contract having a permit with expiry', async (done) => {
+
+    const op = await Tezos.contract.originate({
+      code: permit_admin_42_expiry,
+      storage:
+        {
+        0: 300,
+        1: new MichelsonMap(),
+        2: 0,
+        3: 'tz1h1LzP7U8bNNhow8Mt1TNMxb91AjG3p6KH',
+        4: 'tz1h1LzP7U8bNNhow8Mt1TNMxb91AjG3p6KH'
+        },
+      });
+      await op.confirmation();
+      expect(op.hash).toBeDefined();
+      expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
+      const contract = await op.contract();
+      expect(op.status).toEqual('applied')
+
+      const expiry = 300;
+
+      //submit the expiry to the contract
+       Tezos.contract
+       .at(contract.address)
+       .then((c) => {
+         return c.methods.setExpiry(expiry).send();
+       })
+       .then(async (op) => {
+         await op.confirmation(3);
+         return op.hash;
+       })
+       .catch(Error) ;
+       expect(op.hash).toBeDefined();
+       expect(op.includedInBlock).toBeLessThan(Number.POSITIVE_INFINITY);
```

Comment:
The lines 201 and 202 are testing again the constant op created at line 171 instead of the operation representing the call of the setExpiry method

Reply:
I removed the lines